apps/backend/src/libs/types/repository.type.ts

```diff
@@ -1,8 +1,8 @@
 type Repository<T = unknown> = {
-	create(payload: unknown): Promise<T>;
-	delete(id: number): Promise<boolean>;
-	find(id: number): Promise<T | null>;
-	findAll(): Promise<T[]>;
+	create(payload: unknown): Promise<T | null>;
+	delete(id: number, userId: number): Promise<boolean>;
+	find(id: number, userId: number): Promise<T | null>;
+	findAll(id: number): Promise<T[]>;
```

Comment:
```suggestion
	delete(id: number): Promise<boolean>;
	find(id: number): Promise<T | null>;
	findAll(): Promise<T[]>;
```
you can not modify this things here. since this is a general, for all repositories they should be reusable